### DISCUSSION TITLE
linux 普通用户安装失败的问题，访问不了 /root/.local 目录

### DISCUSSION THREAD

**Discussion by zxhio** [xmake-io/xmake#4287] [General]
xmake 使用时不推荐使用在 root 用户下直接使用，但是在安装时还是有一些目录放在 /root/.local 是不是设计上有一些冲突。
感觉可以像 go 或者 rus 那样，默认就放在 ~/ 目录下。

更新一下操作：
手动编译再进行安装还是有问题，必须要访问 /root/.local/share/xmake/core/_xmake_main.lua 文件，增加 sudo 后表面安装成功，但是实际上 ~/ 目录下是没有东西的。

**Comment by waruqi** [comment]
不会，默认就是全部放的 `~` ，除非你 root 用户登录的，那只能取到 `/root` 作为 `~`

**Reply by zxhio** [reply]
我是 gitlab 的runner用户，这个用户是通过 root su 切换的，难道只能取到 ssh 登录的那个用户嘛，我换了几个方式了，情况是一样的

**Reply by waruqi** [reply]
这边根据 $HOME $TEMDIR 取的，你可以自己检查下，或者自己改下
